diff --git a/node_modules/@greenwood/cli/src/lifecycles/prerender.js b/node_modules/@greenwood/cli/src/lifecycles/prerender.js
index 785c95d..afc5eda 100644
--- a/node_modules/@greenwood/cli/src/lifecycles/prerender.js
+++ b/node_modules/@greenwood/cli/src/lifecycles/prerender.js
@@ -102,7 +102,8 @@ async function preRenderCompilationWorker(compilation, workerPrerender) {
         compilation: JSON.stringify(compilation),
         page: JSON.stringify(page),
         prerender: true,
-        htmlContents: body,
+        // TODO can we try and do this from the standard html resource plugin instead?
+        htmlContents: body.replace(/&#x3C;/g, 'greenwood-custom-left-bracket'),
         scripts: JSON.stringify(scripts)
       }, (err, result) => {
         if (err) {
@@ -117,6 +118,8 @@ async function preRenderCompilationWorker(compilation, workerPrerender) {
       body = body.replace('<!-- greenwood-ssr-start --><!-- greenwood-ssr-end -->', ssrContents);
     }
 
+    body = body.replace(/greenwood-custom-left-bracket/g, '&lt;')
+
     await createOutputDirectory(route, new URL(scratchUrl.href.replace('index.html', '')));
     await fs.writeFile(scratchUrl, body);
 
