diff --git a/node_modules/wc-compiler/src/dom-shim.js b/node_modules/wc-compiler/src/dom-shim.js
index be289a3..2c97d0e 100644
--- a/node_modules/wc-compiler/src/dom-shim.js
+++ b/node_modules/wc-compiler/src/dom-shim.js
@@ -83,6 +83,9 @@ class Document extends Node {
   createDocumentFragment(html) {
     return new DocumentFragment(html);
   }
+
+  querySelector() { }
+  querySelectorAll() { }
 }
 
 // https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement
@@ -102,6 +105,9 @@ class ShadowRoot extends DocumentFragment {
     super();
     this.mode = options.mode || 'closed';
     this.adoptedStyleSheets = [];
+    // TODO not sure if this is the right base class for these?
+    this.querySelector = noop;
+    this.querySelectorAll = noop;
   }
 }
 
